src/services/EventBus.ts

```diff
@@ -30,4 +33,38 @@ EventBus.once = function<Key extends keyof Events>(type: Key, handler: GenericEv
 		this.off(type, fn)
 	}
 	this.on(type, fn)
+
+	// Store reference to the original handler to be able to remove it later
+	if (!_onceHandlers.has(type)) {
+		_onceHandlers.set(type, new Map())
+	}
+	_onceHandlers.get(type)!.set(handler, fn)
+}
+
+const off = EventBus.off.bind(EventBus)
+/**
+ * OVERRIDING OF ORIGINAL MITT FUNCTION
+ * Remove an event handler for the given type.
+ * If `handler` is omitted, all handlers of the given type are removed.
+ * @param type Type of event to unregister `handler` from (`'*'` to remove a wildcard handler)
+ * @param [handler] Handler function to remove
+ */
+EventBus.off = function<Key extends keyof Events>(type: Key, handler?: GenericEventHandler) {
+	// @ts-expect-error: Vue: No overload matches this call
+	off(type, handler)
+
+	const onceHandlers = handler && _onceHandlers.get(type)
+	if (onceHandlers) {
+		for (const [originalFn, fn] of onceHandlers) {
+			if (fn === handler) {
+				onceHandlers!.delete(originalFn)
+			} else if (originalFn === handler) {
+				// @ts-expect-error: Vue: No overload matches this call
+				off(type, fn)
+				onceHandlers!.delete(originalFn)
+			}
+		}
```

Comment:
I don't understand this part. We have a specific handler to remove, why do we need to loop?

Reply:
[fixup](https://github.com/nextcloud/spreed/pull/13769)